### PR TITLE
Add support for socket activation

### LINKED ISSUE #88: Support authentication of nodes
We should be able to define in the manager configuration details about each node, and this should include authentication information for each of the nodes (like a client certification or similar) such that we can verify that the right node is connecting.

### PR DESCRIPTION
**Pull request by ygalblum**
Use the FD passed via the socket activation is exists Otherwise, create the socket according to the configuration flags

Add a socket activation systemd unit and install it

### CODE REVIEW

File: systemd-units/meson.build [hunk 1]
```diff
@@ -7,6 +7,11 @@ install_data(
   install_dir : systemd_unit_dir
 )
 
+install_data(
```
Comment (engelmi) [inline systemd-units/meson.build:10]: I think you need to add this file to the rpm build below [the hirte files](https://github.com/containers/hirte/blob/main/hirte.spec.in#L85)

Comment (ygalblum) [reply systemd-units/meson.build:10]: The PR includes a change in hirte.spec.in. Isn't this what you are referring to?

Comment (engelmi) [reply systemd-units/meson.build:10]: Oh, missed that sneaky file. You are right. Sorry about that.

File: src/manager/manager.c [hunk 2]
```diff
@@ -379,14 +380,27 @@ static int manager_accept_node_connection(
...
-        if (accept_fd < 0) {
-                hirte_log_errorf("Failed to create TCP socket: %s", strerror(-r));
+        int n = sd_listen_fds(0);
+        if (n > 1) {
+                hirte_log_errorf("Received too many file descriptors - %d", n);
                 return false;
         }
 
+        if (n == 1) {
+                accept_fd = SD_LISTEN_FDS_START + 0;
```
Comment (engelmi) [inline src/manager/manager.c]: Is there a reason for adding 0?

**Review by engelmi — APPROVED** [review]
LGTM

File: src/manager/manager.c (hunk 2 above)
Comment (ygalblum) [reply src/manager/manager.c]: You're right. I used an example that has multiple sockets and forgot to cleanup

**Comment by ygalblum** [comment]
Resolves #205

File: systemd-units/hirte.socket [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[Socket]
+ListenStream=2020
```
Comment (alexlarsson) [inline systemd-units/hirte.socket:2]: Gotta remember to change this if we change the default port

**Pull request merged**